File: Chapter 03/C#/Section_3_2_Coupling_Exercise/After/UserReportData.cs
namespace Chapter03.CouplingTest.After;

/// <summary>
/// A Data Transfer Object (DTO).
/// The sole purpose of this class is to be a simple container for data that needs
/// to be passed across an architectural boundary (in this case, from the
/// `UserDataService` to the `UserReportGenerator`).
/// 
/// By creating this "chunky" object, we can reduce the number of calls between
/// the two components from four to just one, which significantly lowers their coupling.
/// </summary>
public class UserReportData
{
    public required string Name { get; set; }
    public required string Email { get; set; }
    public decimal TotalSpent { get; set; }
}

File: Chapter 03/Python/section_3_2_coupling_exercise/after/user_report_data.py
from dataclasses import dataclass

"""
THE DATA TRANSFER OBJECT (DTO).
* ARCHITECTURE NOTE: Using a Python dataclass is the ideal way to implement 
a DTO. It provides a lightweight, immutable-friendly structure that 
acts as a stable contract between the service and its consumers.
"""
@dataclass
class UserReportData:
    name: str
    email: str
    total_spent: float